StellarApp/Protocol/PlanetsWithAction.swift

```diff
@@ -0,0 +1,179 @@
+                orbitNode.addChildNode(planetNode)
+                orbitNode.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: 0.9, z: 0, duration: 1)))
+                planetNodes.append(orbitNode)
+                
+            case .saturn:
+                let orbitNode = SCNNode.getRings(radius: 8, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getPlanetsNode(name: "saturn", image: UIImage(named: "art.scnassets/saturn.jpg")!, radius: 0.7, vector: (x: 8, y: 0, z: 0), moveSpeed: 0.7, rotation: 0.2)
+                let saturnOrbitNode = SCNNode.getSaturnRings(radius: 0.9, vector: (x: 0, y: 0, z: 0), texture: UIImage(named: "art.scnassets/saturnRing.png")!)
+                orbitNode.addChildNode(planetNode)
+                planetNode.addChildNode(saturnOrbitNode)
+                orbitNode.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: 0.7, z: 0, duration: 1)))
+                planetNodes.append(orbitNode)
+                
+            case .uranus:
+                let orbitNode = SCNNode.getRings(radius: 10, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getPlanetsNode(name: "uranus", image: UIImage(named: "art.scnassets/uranus.jpg")!, radius: 0.8, vector: (x: 10, y: 0, z: 0), moveSpeed: 0.5, rotation: 0.4)
+                orbitNode.addChildNode(planetNode)
+                orbitNode.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: 0.5, z: 0, duration: 1)))
+                planetNodes.append(orbitNode)
+                
+            case .neptune:
+                let orbitNode = SCNNode.getRings(radius: 12, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getPlanetsNode(name: "neptune", image: UIImage(named: "art.scnassets/neptune.jpg")!, radius: 0.6, vector: (x: 12, y: 0, z: 0), moveSpeed: 0.4, rotation: 0.8)
+                orbitNode.addChildNode(planetNode)
+                orbitNode.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: 0.4, z: 0, duration: 1)))
+                planetNodes.append(orbitNode)
+                
+            case .pluto:
+                let orbitNode = SCNNode.getRings(radius: 14, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getPlanetsNode(name: "pluto", image: UIImage(named: "art.scnassets/pluto.jpg")!, radius: 0.8, vector: (x: 14, y: 0, z: 0), moveSpeed: 0.4, rotation: 0.8)
+                orbitNode.addChildNode(planetNode)
+                orbitNode.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: 0.4, z: 0, duration: 1)))
+                planetNodes.append(orbitNode)
+                
+            default:
+                break
+            }
+        }
+        return planetNodes
+    }
+    
+    
+}
+
+extension SCNNode {
+    static func deepperScaleNode(node: SCNNode, scaleAction: SCNAction) {
```

Comment:
another set of extension functions on SCNNode. Figure out which ones are being used, delete the rest.